pokemonterminal/platform/named_event/adapters/posix.py

```diff
@@ -0,0 +1,44 @@
+import posix_ipc
+
+from . import NamedEvent
+
+class PosixNamedEvent(NamedEvent):
+    """
+    A wrapper for named events using a named semaphore
+    """
+
+    def __init__(self, name: str):
+        semaphore_name = '/' + name
+        try:
+            self.__semaphore = posix_ipc.Semaphore(semaphore_name, flags=posix_ipc.O_CREX)
+            self.__duplicate = True
+        except posix_ipc.ExistentialError: # Semaphores are reconsidering their life choices
+            self.__semaphore = posix_ipc.Semaphore(semaphore_name)
+            self.__duplicate = False
+
+    # NOTE this doesn't works on macOS
```

Comment:
whats happens on mac? Maybe provide a bit more detail if possible

Reply:
http://semanchuk.com/philip/posix_ipc/#platforms

Reply:
👍 I'd suggest putting that link as a comment right in the code as explanation for why it doesn't work, ie `# NOTE this doesn't works on macOS, see http://semanchuk.com/philip/posix_ipc/#platforms`

Reply:
Yup, I planned to do that.